scout/server/app.py

```diff
@@ -35,6 +35,40 @@
 )
 
 LOG = logging.getLogger(__name__)
+USERS_LOGGER_PATH_PARAM = "USERS_ACTIVITY_LOG_PATH"
+SUB_URL_IGNORE_LIST = [
+    "static",
+    "ideograms",
+    "custom_images",
+    "autozygous_images",
+    "coverage_images",
+    "upd_regions_images",
+    "favicon",
+    "GET /",
+    "Closing",
+    "Enable",
+    "enabled",
+    "Collecting IGV tracks",
+]  # Substrings used when filtering messages to show if users activity log is on
+
+
+class ActivityLogFilter(logging.Filter):
+    """When monitoring users activity, log only navigation on main pages.
+    - Do not log messages that contain the substrings specified in SUB_URL_IGNORE_LIST"""
+
+    def filter(self, record):
+        return any(sub_url in record.getMessage() for sub_url in SUB_URL_IGNORE_LIST) is False
+
+
+def set_activity_log(app):
+    """Log users' activity to a file, if specified in the scout config."""
+    if USERS_LOGGER_PATH_PARAM not in app.config:
+        return
+    app.logger.setLevel("INFO")
```

Comment:
This sets the main log streamhandler to INFO, regardless of the log level asked?

Reply:
yes

Reply:
Ahem, maybe not. This is tricky! 😅